pkgs/development/compilers/gcc/default.nix

```diff
@@ -408,14 +409,28 @@ lib.pipe ((callFile ./common/builder.nix {}) ({
       maintainers
     ;
   } // lib.optionalAttrs (!atLeast11) {
-    badPlatforms = if !is49 then [ "aarch64-darwin" ] else lib.platforms.darwin;
+    badPlatforms = if !(is48 || is49) then [ "aarch64-darwin" ] else lib.platforms.darwin;
   };
-} // optionalAttrs is7 {
-  env.NIX_CFLAGS_COMPILE = lib.optionalString (stdenv.cc.isClang && langFortran) "-Wno-unused-command-line-argument";
+} // optionalAttrs (!atLeast11 && stdenv.cc.isClang) {
+  env.NIX_CFLAGS_COMPILE = toString (
+    lib.optionals (lib.versionAtLeast (lib.getVersion stdenv.cc) "16") [
+      # Silence errors related to use of implicitly declared library functions and implicit ints,
+      # which can cause unwanted `configure` feature detection failures.
+      "-Wno-error=implicit-function-declaration"
+      "-Wno-error=implicit-int"
+    ] ++ lib.optionals (is7 && langFortran) [
+      "-Wno-unused-command-line-argument"
+    ]
+  );
```

Comment:
This change isn't Darwin-specific; please move it to a separate PR.

Reply:
While it’s not Darwin-specific, Darwin is the primary user of clang and was updated to clang 16 this staging cycle. This particular patch was prompted by potential issues I noticed while reviewing `config.log`, but I wasn’t at the point of being able to build GCC at the time, so it may be that it’s not actually necessary. I’m reviewing and building the affected versions to confirm.